Framework/DataObjects/src/PeaksWorkspace.cpp

```diff
@@ -149,6 +150,28 @@ void PeaksWorkspace::removePeak(const int peakNum) {
   peaks.erase(peaks.begin() + peakNum);
 }
 
+/** Removes multiple peaks
+* @param badPeaks peaks to be removed
+*/
+void PeaksWorkspace::removePeaks(const std::vector<int> &badPeaks) {
+  if (badPeaks.size() == 0)
+    return;
+  int i = 0;
+  auto first = peaks.begin();
+  auto end = peaks.end();
+  auto result = first;
+  for (; first < end; ++first, ++i) {
+    // if index of peak is not in badPeaks
+    if (std::find(badPeaks.begin(), badPeaks.end(), i) == badPeaks.end()) {
```

Comment:
As these are integers underneath and the container is arbitrary size could we instead sort then use [std::binary_search](http://en.cppreference.com/w/cpp/algorithm/binary_search)?

This way we do `O(n) + m * O(log N)` operations instead of `m * O(n)` (where m is the number of bad peaks)